test/lib/ansible_test/_internal/core_ci.py

```diff
@@ -303,23 +303,30 @@ def get(self, tries=3, sleep=15, always_raise_on=None):
                 port=self.port or 12345,
                 username='username',
                 password='password' if self.platform == 'windows' else None,
+                response=response,
             )
         else:
             response_json = response.json()
-
             status = response_json['status']
-            con = response_json['connection']
-
-            self.connection = InstanceConnection(
-                running=status == 'running',
-                hostname=con['hostname'],
-                port=int(con.get('port', self.port)),
-                username=con['username'],
-                password=con.get('password'),
-            )
-
-            if self.connection.password:
-                display.sensitive.add(str(self.connection.password))
+            con = response_json.get('connection')
+
+            if con:
+                self.connection = InstanceConnection(
+                    running=status == 'running',
+                    hostname=con['hostname'],
+                    port=int(con.get('port', self.port)),
+                    username=con['username'],
+                    password=con.get('password'),
+                    response=response,
+                )
+            else:  # 'vcenter' resp does not have a 'connection' key
+                self.connection = InstanceConnection(
+                    running=status == 'running',
+                    response=response,
```

Comment:
Return just the response JSON, rather than the entire response.

```suggestion
                    response=response_json,
```

Reply:
Ok, but I name the attribute `response_json` for consistency.